target/mips/helper.c

```diff
@@ -144,6 +139,13 @@ int r4k_map_address(CPUMIPSState *env, hwaddr *physical, int *prot,
 #endif
                     *prot |= PAGE_EXEC;
                 }
+
+#if defined(TARGET_CHERI)
+                if (rw == MMU_DATA_CAP_LOAD && (n ? tlb->L1 : tlb->L0)) {
```

Comment:
Probably doesn't even need to be gated by `MMU_DATA_CAP_LOAD` any more now it's been moved and not touching persistent state (like how `PROT_WRITE` gets set regardless of whether this is actually a store or just a load from a dirty page).

Reply:
That's fair.  This might actually obviate the diff that introduced MMU_DATA_CAP_LOAD, though CAP_STORE is still used.

Reply:
Depending on how you do CLG you may still want it then, and it's probably a good idea to keep the distinction for future use regardless.

Reply:
Good point.  We're being kind of cagey about whether the CLG traps are truly data dependent; I suspect the specification is going to have to allow both answers, in which case, yes, we may want MMU_DATA_CAP_LOAD.